### PR TITLE
stm32f1xx EEPROM emulation

### PR DESCRIPTION
**Pull request by hsgw**
* Add stm32f1xx EEPROM emulation
* Fix eeprom update compare bug

Some EEPROM test code on Maple mini board is here.
https://github.com/hsgw/qmk_firmware/blob/maple_eeprom/keyboards/maple_test/keymaps/eeprom_test/keymap.c

### CODE REVIEW

**Comment by yiancar** [comment]
@skullydazed @jackhumbert  what do you guys say?:)

**Comment by drashna** [comment]
Looks like there is a merge conflict, if you could resolve that?

**Comment by yiancar** [comment]
yep i dnt know when this appeared! @hsgw please?:)

**Comment by drashna** [comment]
Recently, it was a PR of mine that expanded the functionality of the EEPROM stuff, including KB and USER blocks for EEPROM.

File: tmk_core/common/chibios/flash_stm32.c [hunk 1]
```diff
@@ -10,19 +10,30 @@
...
+    #error "not implemented."
+#endif
 
-#include "stm32f3xx.h"
 #include "flash_stm32.h"
 
-#define FLASH_KEY1          ((uint32_t)0x45670123)
-#define FLASH_KEY2          ((uint32_t)0xCDEF89AB)
+#if defined(EEPROM_EMU_STM32F303xC)
+    #define FLASH_KEY1          ((uint32_t)0x45670123)
```
Comment (yiancar) [inline tmk_core/common/chibios/flash_stm32.c]: As the FLASH keys get loaded in the stm32f3xx.h similarly with what you did for F1xx could you take those defines out and use the ones from the header file?:)

Sorry I should have done that at the time

**Comment by drashna** [comment]
@hsgw, if you want, I can fix the merge conflict here.

**Comment by hsgw** [comment]
@drashna I’d like you to fix this. thanks!

**Comment by drashna** [comment]
Okay.  :) 

That should maintain it basically as it was for the updated code.  

If there are any issues with, let me know.

**Comment by hsgw** [comment]
I fixed https://github.com/qmk/qmk_firmware/pull/3914#discussion_r223433276

**Comment by jackhumbert** [comment]
Awesome - thanks!

**Comment by warptozero** [comment]
Somehow this broke my Planck rev6, none of the keys do anything after this commit.

**Comment by jackhumbert** [comment]
@warptozero would you mind making a new issue for this, or hopping on the [QMK Discord](https://discord.gg/fBGYurv)?

**Comment by warptozero** [comment]
@jackhumbert sure, no problem: #4255.

**Pull request merged**